zappa/core.py

```diff
@@ -2930,6 +2936,15 @@ def get_scheduled_event_name(self, event, function, lambda_name, index=0):
             # prefix all entries bar the first with the index
             # Related: https://github.com/Miserlou/Zappa/pull/1051
             name = "{}-{}".format(index, name)
+
+        # https://github.com/zappa/Zappa/issues/1036
+        # Error because the name cannot be obtained if the function name is longer than 64 characters
+        if len(name) >= 64:
+            raise EnvironmentError(
+                "Length Constraints:"
+                "Maximum length of 63 ({{expression index}}-{{event['name']}}?-{{event['function']}}):"
```

Comment:
Ok, this is hard for the user to understand maybe.  

It may be a good idea to include the components used in generating the rule name in the exception, so the user can determine what they need to change.